src/main/java/com/wavefront/generators/TraceGenerator.java

```diff
@@ -86,6 +87,10 @@ protected void startGeneration(boolean isRealTime, @Nonnull GeneratorConfig gene
         }else {
           break;
         }
+        if (AbstractTraceLoader.usedHeapMemory < (double) AbstractTraceLoader.memoryMXBean.getHeapMemoryUsage().getUsed() / 1073741824) {
```

Comment:
I guess that it is better to replace it with a `updateHeapMemory()` method and exclude double calculation. Assign the result to a variable and use it in the second place.